Examples/CaseStudies/SwiftUICaseStudiesTests/01-GettingStarted-SharedStateTests.swift

```diff
@@ -0,0 +1,103 @@
+import Combine
+import ComposableArchitecture
+import XCTest
+
+@testable import SwiftUICaseStudies
+
+class SharedStateTests: XCTestCase {
+  func testTabRestoredOnReset() {
+    let store = TestStore(
+      initialState: SharedState(),
+      reducer: sharedStateReducer,
+      environment: ()
+    )
+    
+    store.assert(
+      .send(.selectTab(.profile)) {
+        $0.currentTab = .profile
+        $0.profile = .init(currentTab: .profile, count: 0, maxCount: 0, minCount: 0, numberOfCounts: 0)
+      },
+      .send(.profile(.resetCounterButtonTapped)) {
+        $0.currentTab = .counter
+        $0.profile = .init(currentTab: .counter, count: 0, maxCount: 0, minCount: 0, numberOfCounts: 0)
+      })
+  }
+
+  func testTabSelection() {
+    let store = TestStore(
+      initialState: SharedState(),
+      reducer: sharedStateReducer,
+      environment: ()
+    )
+
+    store.assert(
+      .send(.selectTab(.profile)) {
+        $0.currentTab = .profile
+        $0.profile = .init(currentTab: .profile, count: 0, maxCount: 0, minCount: 0, numberOfCounts: 0)
+      },
+      .send(.selectTab(.counter)) {
+        $0.currentTab = .counter
+        $0.profile = .init(currentTab: .counter, count: 0, maxCount: 0, minCount: 0, numberOfCounts: 0)
+      })
+  }
+
+  func testSharedCounts() {
+    let store = TestStore(
+      initialState: SharedState(),
+      reducer: sharedStateReducer,
+      environment: ()
+    )
+
+    store.assert(
+      .send(.counter(.incrementButtonTapped)) {
+        $0.counter = .init(alert: nil, count: 1, maxCount: 1, minCount: 0, numberOfCounts: 1)
+        $0.profile = .init(currentTab: .counter, count: 1, maxCount: 1, minCount: 0, numberOfCounts: 1)
+      },
+      .send(.counter(.decrementButtonTapped)) {
+        $0.counter = .init(alert: nil, count: 0, maxCount: 1, minCount: 0, numberOfCounts: 2)
+        $0.profile = .init(currentTab: .counter, count: 0, maxCount: 1, minCount: 0, numberOfCounts: 2)
+      },
+      .send(.counter(.decrementButtonTapped)) {
+        $0.counter = .init(alert: nil, count: -1, maxCount: 1, minCount: -1, numberOfCounts: 3)
+        $0.profile = .init(currentTab: .counter, count: -1, maxCount: 1, minCount: -1, numberOfCounts: 3)
+      })
```

Comment:
Here's another way to do this that might make it clearer how state is changing on a per-field basis:

```suggestion
    store.assert(
      .send(.counter(.incrementButtonTapped)) {
        $0.counter.count = 1
        $0.counter.maxCount = 1
        $0.counter.numberOfCounts = 1
      },
      .send(.counter(.decrementButtonTapped)) {
        $0.counter.count = 0
        $0.counter.numberOfCounts = 2
      },
      .send(.counter(.decrementButtonTapped)) {
        $0.counter.count = -1
        $0.counter.minCount = -1
        $0.counter.numberOfCounts = 3
      })
```

Reply:
That is more readable, though the downside is that it doesn't mention the `profile` at all, so it may not be clear to readers that the changes to counter are also expected to change the profile.